pynamodb/models.py

```diff
@@ -344,9 +357,14 @@ def delete(self, condition=None, conditional_operator=None, **expected_values):
             kwargs.update(expected=self._build_expected_values(expected_values, DELETE_FILTER_OPERATOR_MAP))
         kwargs.update(conditional_operator=conditional_operator)
         kwargs.update(condition=condition)
+
+        if in_transaction is not None:
+            operation_kwargs = self._get_connection().get_operation_kwargs_for_delete_item(*args, **kwargs)
+            in_transaction.add_delete_item(operation_kwargs)
+            return True
```

Comment:
The return statements here (and for other operation results) seem kinda odd - we're really entering a promise state, rather than a straightforward result return. I'm not sure returning a default is a good eventual form

Reply:
fwiw, overloaded return types might actually be possible with mypy literals: https://mypy.readthedocs.io/en/latest/literal_types.html
e.g.
```py
def delete(..., in_transaction=Literal[None], ...) -> ...:
```

It is somewhat odd but 🤷‍♂ I guess that's why Python is not Java?

Reply:
ok, i will attempt to overload :0

Reply:
Sorry, of course I meant:
```py
@overload
def delete(..., in_transaction: None, ...) -> ...:
@overload
def delete(..., in_transaction: Transaction, ...) -> ...:
```
(`:` and not `=`)